A89-xds-cluster-metric-label.md

```diff
@@ -0,0 +1,75 @@
+A89: xDS Cluster Metric Label
+----
+* Author(s): [Eric Anderson](https://github.com/ejona86)
+* Approver: @markdroth
+* Status: {Draft, In Review, Ready for Implementation, Implemented}
+* Implemented in: <language, ...>
+* Last updated: 2025-01-10
+* Discussion at: <google group thread> (filled after thread exists)
+
+## Abstract
+
+Add a new optional label to per-call metrics containing the xDS cluster being
+used for the RPC.
+
+## Background
+
+[gRFC A78][A78] added the `grpc.lb.locality` per-call optional label, which also
+added the infrastructure to support LBs adding optional labels to per-call
+metrics. The optional label can be enabled in the gRPC/OpenTelemetry integration
+API added in [gRFC A79][gRFC A79].
+
+Similar to how locality metrics are useful for analyzing _where_ traffic is
+being routed, the xDS cluster is useful for knowing _to whom_ it is being
+routed. `grpc.target` is generally all that's necessary to know which service is
+receiving traffic, but non-deterministic routing in xDS like weighted clusters,
+aggregate clusters, and cluster specifier plugins mean different clusters (and
+thus potentially different services or service versions) would comingle metrics
+unless the selected cluster is added as a label. It can also be helpful to know
+the selected cluster to confirm that deterministic routing like path matching is
+behaving as expected.
+
+### Related Proposals:
+* [gRFC A78: gRPC OTel Metrics for WRR, Pick First, and XdsClient][gRFC A78]
+* [gRFC A79: Non-per-call Metrics Architecture][gRFC A79]
+
+[gRFC A78]: A78-grpc-metrics-wrr-pf-xds.md#per-call-metrics
+[gRFC A79]: A79-non-per-call-metrics-architecture.md
+
+## Proposal
+
+Each pick in the `xds_clunter_impl` policy, `xds_cluster_impl` will add the
+optional label `grpc.xds.cluster` to the call attempt tracer. The value will be
+copied from `xds_clunter_impl`'s service config `cluster` key. This is done
+regardless of the pick's result. It is possible for later picks for the same RPC
+to have a different value. This is the case for locality as well, and the last
+pick's value should be used.
+
+The `grpc.xds.cluster` label will be available on the following per-call
```

Comment:
@yashykt, [Java hard-codes](https://github.com/grpc/grpc-java/blob/d65d3942e6078c75500b034155a58b3fa6d516e8/opentelemetry/src/main/java/io/grpc/opentelemetry/OpenTelemetryMetricsModule.java#L209) support for locality today. I remember there being a conversation about supporting any optional labels, and I think we had kicked that can down the road.

I'm envisioning just adding another hard-coded case to the otel module for this.